core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -58,7 +58,7 @@ import java.util.Arrays
   * Rewinding tends to make error reporting more difficult and can lead
   * to exponential parser behavior it is not the default behavior.
   */
-sealed abstract class Parser0[+A] {
+sealed abstract class Parser0[+A] extends Product {
```

Comment:
Can we do `{ self: Product =>` in abstract class? I forget. That may allow us to bypass the binary compact problem since all subclasses are Products.

Reply:
Seems like it works, if I add it to `Parser` as well.